src/teststat/BinnedNLLH.cpp

```diff
@@ -221,13 +216,13 @@ BinnedNLLH::RegisterFitComponents(){
     fComponentManager.AddComponent(&fPdfManager);
     
     //Because the limits are set by name they can be added in any order.
-    const std::map<std::string, std::vector<Systematic*> > sys_ = fSystematicManager.GetSystematicsGroup();
+    const auto sys_ = fSystematicManager.GetSystematicsGroup();
```

Comment:
The argument for using auto is that it de-clutters what is a rather complicated variable type definition, because the type is already implicitly known from the definition of `GetSystematicsGroup()`.

It's only a style choice; your argument for making the type explicit is also reasonable so I'm happy to revert it for thesse cases.